src/main/java/com/salesforce/functions/jvm/sdk/bulk/BulkApi.java

```diff
@@ -46,11 +44,11 @@ public interface BulkApi {
    *     href="https://developer.salesforce.com/docs/atlas.en-us.api_asynch.meta/api_asynch/get_job_successful_results.htm">Bulk
    *     API 2.0 and Bulk API Developer Guide</a>
    * @param jobId The ID of the job to get successful record results for.
-   * @return A CSV file with successful records.
+   * @return An iterable containing the records that were successfully processed.
    * @throws BulkApiException If an error occurred while querying successful record results.
    */
   @SuppressWarnings("unused")
-  File getJobSuccessfulRecordResults(String jobId) throws BulkApiException;
+  Iterable<Record> getJobSuccessfulRecordResults(String jobId) throws BulkApiException;
```

Comment:
Returns an `Iterable<Record>` so that the result can be used in `for each` statements and it could be backed by a custom iterator that keeps memory usage low when large data sets are involved.